src/main/java/org/sonarlint/intellij/ui/HotspotCellRenderer.java

```diff
@@ -0,0 +1,86 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonarlint.intellij.ui;
+
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.ui.JBColor;
+import org.sonarlint.intellij.issue.hotspot.LocalHotspot;
+import org.sonarlint.intellij.ui.nodes.HotspotNode;
+import org.sonarsource.sonarlint.core.client.api.connected.RemoteHotspot.Rule.Probability;
+
+import javax.swing.BorderFactory;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JTree;
+import javax.swing.SwingConstants;
+import javax.swing.border.Border;
+import javax.swing.tree.TreeCellRenderer;
+import java.awt.Component;
+import java.awt.FlowLayout;
+import java.awt.Font;
+import java.util.EnumMap;
+import java.util.Map;
+
+class HotspotCellRenderer implements TreeCellRenderer {
+
+  private static final int HORIZONTAL_PADDING = 15;
+
+  private static final Map<Probability, JBColor> colorsByProbability = new EnumMap<>(Probability.class);
+
+  private static final int RED = 0xd4333f;
+  private static final int ORANGE = 0xed7d20;
+  private static final int YELLOW = 0xeabe06;
+  private static final int WHITE = 0xffffff;
+
+  static {
+    colorsByProbability.put(Probability.HIGH, new JBColor(RED, RED));
+    colorsByProbability.put(Probability.MEDIUM, new JBColor(ORANGE, ORANGE));
+    colorsByProbability.put(Probability.LOW, new JBColor(YELLOW, YELLOW));
+  }
+
+  @Override
+  public Component getTreeCellRendererComponent(JTree jTree, Object value, boolean selected, boolean expanded, boolean leaf, int row, boolean hasFocus) {
+    LocalHotspot hotspot = ((HotspotNode) value).getHotspot();
+    FlowLayout layout = new FlowLayout(FlowLayout.LEADING);
+    JPanel panel = new JPanel(layout);
+
+    JLabel probabilityLabel = new JLabel(hotspot.getProbability().name());
+    Border border = BorderFactory.createEmptyBorder(0, HORIZONTAL_PADDING, 0, HORIZONTAL_PADDING);
+    probabilityLabel.setBorder(border);
+    probabilityLabel.setVerticalTextPosition(SwingConstants.TOP);
+    probabilityLabel.setBackground(colorsByProbability.get(hotspot.getProbability()));
+    probabilityLabel.setForeground(new JBColor(WHITE, WHITE));
+    probabilityLabel.setOpaque(true);
+    probabilityLabel.setFont(probabilityLabel.getFont().deriveFont(probabilityLabel.getFont().getStyle() | Font.BOLD));
+    panel.add(probabilityLabel);
+
+    panel.add(new JLabel(hotspot.getMessage()));
+
+    VirtualFile file = hotspot.primaryLocation.file;
+    JLabel fileNameLabel = new JLabel();
+    panel.add(fileNameLabel);
+
+    Integer lineNumber = hotspot.getLineNumber();
+    JLabel lineLabel = new JLabel(file.getName() + (lineNumber != null ? (":" + lineNumber.toString()) : ""));
```

Comment:
Same. No need in parentheses inside ternary operator.

Reply:
See previous comment